oneflow/user/kernels/onednn_pool_kernel_util.h

```diff
@@ -0,0 +1,112 @@
+      ep::Stream* stream, const dnnl::memory::dims src_dims, const dnnl::memory::dims dst_dims,
+      const dnnl::memory::dims kernel_dims, const dnnl::memory::dims strides_dims,
+      const dnnl::memory::dims padding_dims_l, const dnnl::memory::dims padding_dims_r,
+      const dnnl::memory::dims dilation, const void* src, void* dest, int64_t* indice_ptr) {
+    auto data_type = CppTypeToOneDnnDtype<T>();
+    ep::CpuStream* cpu_stream = stream->As<ep::CpuStream>();
+    size_t num_threads = cpu_stream->device()->GetNumThreads();
+    ep::CpuNumThreadsGuard guard(num_threads);
+    dnnl::engine* onednn_engine = cpu_stream->onednn_engine();
+    dnnl::stream* onednn_stream = cpu_stream->onednn_stream();
+
+    auto src_md = dnnl::memory::desc(src_dims, data_type, dnnl::memory::format_tag::nchw);
+    auto dst_md = dnnl::memory::desc(dst_dims, data_type, dnnl::memory::format_tag::nchw);
+    auto src_mem = dnnl::memory(src_md, *onednn_engine, (void*)src);
+    auto dst_mem = dnnl::memory(dst_md, *onednn_engine, (void*)dest);
+
+    auto pooling_d = dnnl::pooling_v2_forward::desc(
+        dnnl::prop_kind::forward_training, dnnl::algorithm::pooling_max, src_md, dst_md,
+        strides_dims, kernel_dims, dilation, padding_dims_l, padding_dims_r);
+    auto pooling_pd = dnnl::pooling_v2_forward::primitive_desc(pooling_d, *onednn_engine);
+    auto pooling_prim = dnnl::pooling_v2_forward(pooling_pd);
+    auto workspace_mem =
+        dnnl::memory(pooling_pd.workspace_desc(), *onednn_engine, (void*)indice_ptr);
+
+    pooling_prim.execute(
+        *onednn_stream,
+        {{DNNL_ARG_SRC, src_mem}, {DNNL_ARG_DST, dst_mem}, {DNNL_ARG_WORKSPACE, workspace_mem}});
+    onednn_stream->wait();
+  }
+
+  static void OneDnnpool1dBackwardCompute(
+      ep::Stream* stream, const dnnl::memory::dims diff_dst_dims,
+      const dnnl::memory::dims diff_src_dims, const dnnl::memory::dims kernel_dims,
+      const dnnl::memory::dims strides_dims, const dnnl::memory::dims padding_dims_l,
+      const dnnl::memory::dims padding_dims_r, const dnnl::memory::dims dilation, void* diff_dst,
+      void* diff_src, void* workspace) {
+    auto data_type = CppTypeToOneDnnDtype<T>();
+    ep::CpuStream* cpu_stream = stream->As<ep::CpuStream>();
+    size_t num_threads = cpu_stream->device()->GetNumThreads();
+    ep::CpuNumThreadsGuard guard(num_threads);
+    dnnl::engine* onednn_engine = cpu_stream->onednn_engine();
+    dnnl::stream* onednn_stream = cpu_stream->onednn_stream();
+
+    auto diff_dst_md = dnnl::memory::desc(diff_dst_dims, data_type, dnnl::memory::format_tag::nchw);
+    auto diff_src_md = dnnl::memory::desc(diff_src_dims, data_type, dnnl::memory::format_tag::nchw);
+    auto diff_dst_mem = dnnl::memory(diff_dst_md, *onednn_engine, diff_dst);
+    auto diff_src_mem = dnnl::memory(diff_src_md, *onednn_engine, diff_src);
+
+    auto pooling_back_d = dnnl::pooling_v2_backward::desc(dnnl::algorithm::pooling_max, diff_src_md,
```

Comment:
```suggestion
    auto pooling_back_desc = dnnl::pooling_v2_backward::desc(dnnl::algorithm::pooling_max, diff_src_md,
```

_d 这个后缀不是很好，onednn 示例代码里也是用 _desc 后缀

Reply:
已改
https://github.com/oneapi-src/oneDNN/blob/master/examples/primitives/pooling.cpp#L105
onednn 示例代码里是_d

Reply:
确实，看来不同的示例代码用的后缀不一样